src/components/tooltip/Tooltip.tsx

```diff
@@ -66,35 +66,41 @@ const positions = {
     },
 };
 
-export interface TooltipProps {
+type DefaultTooltipProps = {
+    /** Whether to constrain the tooltip to the element's scroll parent. Defaults to `false` */
+    constrainToScrollParent: boolean;
+    /** Whether to constrain the tooltip to window. Defaults to `true` */
+    constrainToWindow: boolean;
+    /** Forces the tooltip to be disabled irrespecitve of it's shown state. Defaults to `false` */
+    isDisabled: boolean;
+    /** Where to position the tooltip relative to the wrapped component */
+    position: TooltipPosition | TooltipCustomPosition;
+    /** Tooltip theme */
+    theme: TooltipTheme;
+};
+
+export type TooltipProps = {
     bodyElement?: HTMLElement;
     /** A React element to put the tooltip on */
     children: React.ReactChild;
     /** A CSS class for the tooltip */
     className?: string;
-    /** Whether to constrain the tooltip to the element's scroll parent. Defaults to `false` */
-    constrainToScrollParent?: boolean;
-    /** Whether to constrain the tooltip to window. Defaults to `true` */
-    constrainToWindow?: boolean;
-    /** Forces the tooltip to be disabled irrespecitve of it's shown state. Defaults to `false` */
-    isDisabled?: boolean;
     /** Forces the tooltip to be shown or hidden (useful for errors) */
     isShown?: boolean;
     /** Whether to add tabindex=0.  Defaults to `true` */
     isTabbable?: boolean;
     /** Function called if the user manually dismisses the tooltip - only applies if showCloseButton is true */
     onDismiss?: () => void;
-    /** Where to position the tooltip relative to the wrapped component */
-    position?: TooltipPosition | TooltipCustomPosition;
     /** Shows an X button to close the tooltip. Useful when tooltips are force shown with the isShown prop. */
     showCloseButton?: boolean;
     /** stop click|keypress event bubbling */
     stopBubble?: boolean;
     /** Text to show in the tooltip */
     text?: React.ReactNode;
-    /** Tooltip theme */
-    theme?: TooltipTheme;
-}
+} & Partial<DefaultTooltipProps>;
+// Fixes issue where webpack thinks TooltipProps is undefined due to how Typescript types are removed
+// when compiling.
+export const TooltipProps = undefined;
```

Comment:
seeing some issues importing `TooltipProps`:

```
WARNING in ./src/components/tooltip/index.ts 1:0-81
"export 'TooltipProps' was not found in './Tooltip'
 @ ./src/icons/icons.stories.tsx
 @ ./src sync \.stories\.(js|tsx)$
 @ ./.storybook/config.js
 @ multi ./node_modules/@storybook/core/dist/server/common/polyfills.js ./node_modules/@storybook/core/dist/server/preview/globals.js ./node_modules/@storybook/addon-docs/dist/frameworks/common/config.js ./node_modules/@storybook/addon-docs/dist/frameworks/react/config.js ./.storybook/config.js ./node_modules/webpack-hot-middleware/client.js?reload=true&quiet=true
```

I think it's an issue with webpack and the TS interfaces. Need to explore this more. Adding this did resolve the issue, I think it removes `TooltipProps` as an export when webpack runs?